apps/dashboard/src/app/team/[team_slug]/(team)/~/engine/(instance)/[engineId]/_components/version.tsx

```diff
@@ -18,37 +10,73 @@ import {
   DialogHeader,
   DialogTitle,
 } from "@/components/ui/dialog";
+import {
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { ToolTipLabel } from "@/components/ui/tooltip";
 import { TrackedLinkTW } from "@/components/ui/tracked-link";
+import {
+  type EngineInstance,
+  useEngineGetDeploymentPublicConfiguration,
+  useEngineSystemHealth,
+  useEngineUpdateDeployment,
+} from "@3rdweb-sdk/react/hooks/useEngine";
+import { formatDistanceToNow } from "date-fns";
+import {
+  CircleArrowDownIcon,
+  CloudDownloadIcon,
+  TriangleAlertIcon,
+} from "lucide-react";
+import { useState } from "react";
 import { toast } from "sonner";
 import invariant from "tiny-invariant";
 
 export const EngineVersionBadge = ({
   instance,
+  teamId,
 }: {
   instance: EngineInstance;
+  teamId: string;
 }) => {
   const healthQuery = useEngineSystemHealth(instance.url);
-  const latestVersionQuery = useEngineLatestVersion();
+  const publicConfigurationQuery = useEngineGetDeploymentPublicConfiguration({
+    teamId,
+  });
   const [isModalOpen, setModalOpen] = useState(false);
 
-  const currentVersion = healthQuery.data?.engineVersion ?? "...";
-  const latestVersion = latestVersionQuery.data;
-  const isStale = latestVersion && currentVersion !== latestVersion;
+  if (
+    !healthQuery.data ||
+    publicConfigurationQuery.data.serverVersions?.length === 0
```

Comment:
Build is failing. Type issue: 
https://vercel.com/thirdweb/thirdweb-www/C2i1x6cZEVJuKfQSe6uz45NXdhnv?filter=errors